Bump pulsar client to latest version

spacid opened the pull request:
There is occasionally a Pulsar producer connection error which isn't recoverable. Use the latest Pulsar client in the hope that the connection gets re-established.

Furthermore:
 - Bump some other dependecies to ensure successful build.
 - Use compatible Rust version (=latest) in Docker.

spacid committed:
Bump pulsar client to latest version

There is occasionally a Pulsar producer connection error which isn't recoverable.
Use the latest Pulsar client in the hope that the connection gets re-established.

Furthermore:
 - Bump some other dependecies to ensure successful build.
 - Use compatible Rust version (=latest) in Docker.

spacid commented:
In het algemeen, de versies van de dependecies kan eigenlijk beter:
 - Indien stable (major >= 1): De huidige major versie kiezen bv: `actix-web = "4"`
 - Indien 0-ver (major = 0): De huige minor kiezen bv: envy = "0.4"

Wdyt @maartends ?

maartends reviewed (approved):
Sounds like good versioning practice! This way, we're compatible with [Cargo's interpretation of SemVer](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html). :+1:

spacid committed:
Use consistent SemVer in dependencies

Change versions of dependencies to be more consistent:
- If stable (major >=1 ): Use current major, for example: `actix-web = "4"`.
- if 0-ver (major = 0): Use current minor, for example: `envy = "0.4"`.
